Heap corruption while running cswinrt.exe

While generating projections using cswinrt.exe, sometimes it there's a random heap corruption failure as during [this build ](https://dev.azure.com/microsoft/Dart/_build/results?buildId=36743073&view=logs&j=3ad9154c-b66f-5e48-f703-44743de98f45&t=89909ce3-37d7-533c-75fb-a5d7a53016c2&l=8147)

```  
D:\a\1\s\nuget\Microsoft.Windows.CsWinRT.targets(112,5): error MSB3073: The command ""D:\a\1\s\src\_build\x64\Release\cswinrt\bin\cswinrt.exe" @"D:\a\1\s\src\Projections\WinUI\Generated Files\net5.0\cswinrt.rsp"" exited with code -1073740940. [D:\a\1\s\src\Projections\WinUI\WinUI.csproj]
```

--- a/src/cswinrt/code_writers.h
+++ b/src/cswinrt/code_writers.h
@@ -6536,7 +6536,8 @@ bind<write_event_invoke_args>(invokeMethodSig));
                     {
                         auto&& eventTypeSemantics = get_type_semantics(eventObj.EventType());
                         auto&& eventTypeCode = w.write_temp("%", bind<write_type_name>(eventTypeSemantics, typedef_name_type::Projected, false));
-                        typeNameToDefinitionMap[eventTypeCode] = w.write_temp("%", bind<write_event_source_subclass>(eventTypeSemantics));
+                        auto&& eventClass = w.write_temp("%", bind<write_event_source_subclass>(eventTypeSemantics));
+                        typeNameToDefinitionMap[eventTypeCode] = eventClass;
                     }
                 });
         }
@@ -6548,7 +6549,8 @@ bind<write_event_invoke_args>(invokeMethodSig));
         {
             auto&& eventTypeSemantics = get_type_semantics(eventObj.EventType());
             auto&& eventTypeCode = w.write_temp("%", bind<write_type_name>(eventTypeSemantics, typedef_name_type::Projected, false));
-            typeNameToDefinitionMap[eventTypeCode] = w.write_temp("%", bind<write_event_source_subclass>(eventTypeSemantics));
+            auto&& eventClass = w.write_temp("%", bind<write_event_source_subclass>(eventTypeSemantics));
+            typeNameToDefinitionMap[eventTypeCode] = eventClass;
         }
     }
 }

